### PR TITLE
Some docs for bin_io

### PR DESCRIPTION
**Pull request by psteckler**
Describe some features we use for `bin_io` serialization, that may not be obvious.

### CODE REVIEW

File: docs/specs/bin_io/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+(* containers for previous functions *)
+val bin_writer_t : Bin_spec.t Bin_prot.Type_class.writer
+val bin_reader_t : Bin_spec.t Bin_prot.Type_class.reader
+val bin_t : Bin_spec.t Bin_prot.Type_class.t
+```
+If writing `Bin_prot`-compatible code in a language other than OCaml, that
+code would have to replicate the serialization and deserialization strategy
+used in the custom code.
+
+`bin_size_t` is used to  `bin_write_t` is used to . `bin_read_t` is used
```
Comment (bkase) [inline docs/specs/bin_io/README.md:87]: This looks like some sentence fragments

**Review by bkase — COMMENTED** [review]
This is great!

File: docs/specs/bin_io/README.md (hunk 1 above)
Comment (bkase) [reply docs/specs/bin_io/README.md:87]: Will you flesh this paragraph out before we land it?

Comment (psteckler) [reply docs/specs/bin_io/README.md]: oops, I had moved that text to the comments.

**Review by bkase — APPROVED** [review]

**Comment by bkase** [comment]
!approved-for-mainnet

**Pull request merged**